alibi_detect/od/base.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Dict, Any, Union
+
+from typing_extensions import Protocol, runtime_checkable
+import numpy as np
+
+from alibi_detect.base import BaseDetector
+
+
+class OutlierDetector(BaseDetector, ABC):
+    threshold_inferred = False
+
+    def __init__(self):
+        """ Base class for outlier detection algorithms."""
+        super().__init__()
+        self.meta['online'] = False
+        self.meta['detector_type'] = 'outlier'
+
+    @abstractmethod
+    def fit(self, x: np.ndarray) -> None:
```

Comment:
Noting that the `fit` and `infer_threshold` abstract methods are currently added to outlier (and adversarial) detectors via the `FitMixin` and `ThresholdMixin`. Seems like minor duplication to redefine these methods in `OutlierDetector` rather than subclassing these Mixins? We could also possibly define the `NotFitException` and `ThresholdNotInferredException` exceptions in the Mixin?  This is a minor point as we're not heavily focussed on adversarial detectors anyway...

Reply:
Yep, I've moved the exceptions to the `base.py` class and removed the `OutlierDetector` base class. After removing the fit and threshold method and abstracting the backend functionality out the base class became somewhat redundant!